### ISSUE TITLE
Working Cypher Query in 3.2.3 broken in 3.3.2

### ISSUE THREAD

**Issue by billscheidel** [neo4j/neo4j#10909]
I recently started testing against 3.3.2 and found a query that is now failing.  

The following query produces 1 result in 3.2.3 but 0 results in 3.3.2.  Basically I'm looking for a set of nodes with a particular set of labels.

### Steps to reproduce

- Neo4j version: 3.3.2
- Operating system: Ubuntu 16.04
- API/Driver: Browser using Bolt

```
CREATE (:Foo { id: 1 });
CREATE (:Bar { id: 1 });

MATCH (n { id: 1 }) WHERE LABELS(n) IN ['Foo', 'Baz'] RETURN n;
```

### Expected behavior

1 result returned:  (:Foo)

### Actual behavior

0 results

**Comment by billscheidel** [comment]
Looks like the following query produces the expected results:

```
MATCH (n { id: 1 }) WHERE ANY(x IN LABELS(n) WHERE x IN ['Foo']) RETURN n;
```

**Comment by Lojjs** [comment]
@billscheidel thanks for reporting. I have been able to reproduce it and as you say it is a bug in 3.3 but working as expected in 3.2. I will come back to you once we have solved it.

Best regards Louise, Neo4j Cypher team

**Comment by pontusmelke** [comment]
Hi @billscheidel 
You are right that it works in 3.2.3, however it shouldn't have - sorry about that 😞 

`LABELS(n)` returns a list so you are basically checking if a list is contained in another list which should evaluate to `false` in this case but unfortunately didn't work as intended in 3.2.3 but has been fixed in later versions of 3.2 (and in 3.3).

The canonical way of doing this is by using

```
MATCH (n { id: 1 }) WHERE n:Foo OR n:Baz RETURN n;
```

Sorry for the inconvenience and thanks for reporting!
Regards
Pontus
